core/services/pg/lease_lock.go

```diff
@@ -197,7 +205,7 @@ func (l *leaseLock) loop(ctx context.Context) {
 		case <-stats.C:
 			publishStats(l.db.Stats())
 		case <-ctx.Done():
-			qctx, cancel := DefaultQueryCtx()
+			qctx, cancel := context.WithTimeout(context.Background(), l.cfg.DatabaseDefaultQueryTimeout())
```

Comment:
ok, this one should be easy to fix

Reply:
I am not comfortable changing the behavior of critical systems in this PR. This should addressed separately and with confirmation from @samsondav that it is not by design.